tests/README.md

```diff
@@ -86,27 +49,79 @@ service (or your `QGIS_PGTEST_DB` connection string) uses a connection
 method which allows passing username/password pair to connect (and
 which allows creating users)
 
-# Write tests
+Some tests require a specific PostgreSQL server configuration
+(installation of auth certificates) so the most convenient way
+to bring up such server would be to:
 
-Instructions about writing tests for the processing framework
-can be found in a [separate README file](../python/plugins/processing/tests/README.md):
+    GH_WORKSPACE=$PWD \
+    docker-compose -f .docker/docker-compose-testing-postgres.yml up -d postgres
+    export PGHOST=`docker inspect docker_postgres_1 | jq -r .[0].NetworkSettings.Networks.docker_default.IPAddress`
+    export PGUSER=docker
+    export PGPASSWORD=docker
+    tests/testdata/provider/testdata_pg.sh
+    echo "${PGHOST} postgres # for qgis_test, docker" | sudo tee -a /etc/hosts
 
-    ${TOP_SRCDIR}/python/plugins/processing/tests/README.md
 
-Information about labeling tests design and organization:
+# Running the tests
 
-    ${TOP_SRCDIR}/tests/testdata/labeling/README.rst
+Once the test environment is setup, you can run all tests using `make check`.
+Note you will need `xvfb-run` for that (sudo apt-get install xvfb).
 
-WCS testing information can be found in:
+Individual tests can be run using `ctest`.
 
-    ${TOP_SRCDIR}/tests/testdata/raster/README.WCS
+For example if the output of `make check` ends like this:
 
-About benchmark tests you can read:
+```
+   The following tests FAILED:
+         77 - PyQgsLocalServer (Failed)
+```
 
-    ${TOP_SRCDIR}/tests/bench/README
+You could re-run the failing test with:
 
+```
+   ctest -V -R PyQgsLocalServer
+```
 
-# Run python tests in GDB
+The parameter `-V` enables verbose mode and `-R` takes a regular expression as
+parameter and will only run matching tests.
+
+Note that the name of the test as reported by `make check` does not
+always matches the name of the test as known by `cmake`. This is
+pretty annoying, but true. When this happens, the `ctest -R <guessedName>`
```

Comment:
I don't think we should be putting personal judgment about the state into the readme

Reply:
Statement removed in 7c3fe5caca1